### PR TITLE
[GH workflows] Execute the integration test suite directly on the VM runner & run the simple PyPI test server as a service

### PR DESCRIPTION
**Pull request by eskultety**
This PR is effectively just a vol.1 in a series of small refactorings of our GH CI.
This one only takes care of how:
- we run the integration test suite
- we spin up helper services

This PR **does not** tweak how logs are captured from GH workflows, those remain messy and will handled as part of a different work.

# Maintainers will complete the following section

- [ ] Commit messages are descriptive enough
- [ ] Code coverage from testing does not decrease and new code is covered
- [ ] Docs updated (if applicable)
- [ ] Docs links in the code are still valid (if docs were updated)

**Note:** if the contribution is external (not from an organization member), the CI
pipeline will not run automatically. After verifying that the CI is safe to run:

- [approve GitHub Actions workflows][approve-gh-actions] by clicking a button
- approve the Red Hat Trusted App Pipeline container build by commenting `/ok-to-test`
  (as is the [standard for Pipelines as Code][pac-running-pipeline])

[approve-gh-actions]: https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks
[pac-running-pipeline]: https://pipelinesascode.com/docs/guide/running/#running-the-pipelinerun

### CODE REVIEW

File: .github/workflows/gating.yaml [hunk 1]
```diff
@@ -91,7 +91,7 @@ jobs:
 
   build-image:
     name: Build Cachi2 image and run integration tests on it
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
Comment (a-ovchinnikov) [inline .github/workflows/gating.yaml:94]: Will a TODO comment here help not to forget to revert it?

**Review by a-ovchinnikov — APPROVED** [review]
LGTM!

**Comment by eskultety** [comment]
FWIW: here's a workflow run on my own fork with #580 changes applied as well, so looks like this works with more services in the background nicely.

File: .github/workflows/gating.yaml [hunk 2]
```diff
@@ -104,6 +104,15 @@ jobs:
         with:
           fetch-depth: 0
 
+      - name: PyPI test server start
```
Comment (MartinBasti) [inline .github/workflows/gating.yaml:107]: Shouldn't we just start it right before integration tests?

File: tests/integration/conftest.py [hunk 3]
```diff
@@ -47,7 +47,11 @@ def cachi2_image() -> utils.Cachi2Image:
 # test output.
 @pytest.fixture(autouse=True, scope="session")
 def local_pypiserver() -> Iterator[None]:
-    if os.getenv("CACHI2_TEST_LOCAL_PYPISERVER") != "true":
+    if (
+        os.getenv("CI")
+        and os.getenv("GITHUB_ACTIONS")
```
Comment (slimreaper35) [inline tests/integration/conftest.py:52]: Both variables are needed ?

Comment (eskultety) [reply tests/integration/conftest.py:52]: Technically no, but IMO decreases the likelyhood of a conflict, i.e. a false positive

File: .github/workflows/gating.yaml (hunk 2 above)
Comment (eskultety) [reply .github/workflows/gating.yaml:107]: Does it really matter when we start it?

Comment (eskultety) [reply .github/workflows/gating.yaml:110]: To support my argument a bit, the server usually takes a couple of seconds to spin up, so if a container fails to execute (it's CI after all) I want to fail early and not wait until the cachi2 image is built, so I prefer spinning up the servers as soon as possible in the build & integration job.

Comment (MartinBasti) [reply .github/workflows/gating.yaml:110]: Running process that takes resources when you don't need it matters to me, but I assume it's small.

Comment (eskultety) [reply .github/workflows/gating.yaml:110]: Sure, but running a container image build takes them as well and the image is useless if a test server which is required isn't running ;) , so I'd argue the lesser evil wins -> marking as resolved.

**Review by MartinBasti — APPROVED** [review]

**Review by slimreaper35 — APPROVED** [review]

**Pull request merged**